app/src/main/java/com/example/campusguide/DirectionsActivity.kt

```diff
@@ -29,8 +27,8 @@ class DirectionsActivity : AppCompatActivity() {
     private lateinit var end: String
     private lateinit var startName: String
     private lateinit var endName: String
-    private var travelMode = "Driving"
-    private lateinit var path: PathPolyline
+    private lateinit var currentPath: PathPolyline
```

Comment:
Are all three travel modes computed once and then the radio button only swaps which one is displayed?

Reply:
Yep, that's exactly how it works